salt/modules/dnsmasq.py

```diff
@@ -176,11 +204,9 @@ def _parse_dnamasq(filename):
                     if isinstance(fileopts[comps[0]], str):
                         temp = fileopts[comps[0]]
                         fileopts[comps[0]] = [temp]
-                    fileopts[comps[0]].append(comps[1].strip())
+                    fileopts[comps[0]].append(comps[1])
                 else:
-                    fileopts[comps[0]] = comps[1].strip()
+                    fileopts[comps[0]] = comps[1]
```

Comment:
This will no longer remove whitespace between the equals sign and the key/value like `key = value`. Since we have already stripped the beginning and end of the string we can call just rstrip on the key and lstrip on the value or change to use a regex split like `re.split("\s*=\s*")`. Also we should set the max split on the split in case the value has a `=` in it.

```python
                comps = line.split("=", 1)
                key = comps[0].rstrip()
                if key in fileopts:
                    if isinstance(fileopts[key], str):
                        temp = fileopts[key]
                        fileopts[key] = [temp]
                    fileopts[key].append(comps[1].lstrip())
                else:
                    fileopts[key] = comps[1].lstrip()
```
```
                   fileopts[comps[0]].append(comps[1])
                else:
                    fileopts[comps[0]] = comps[1]
```